File: AlgorithmStudy/Programmers/하노이의탑.js
function solution(n) {
  const answer = [];
  function hanoi(num, from, to, other) {
    if (num === 0) return;
    hanoi(num - 1, from, other, to);
    answer.push([from + 1, to + 1]);
    hanoi(num - 1, other, to, from);
  }
  hanoi(n, 0, 2, 1);
  return answer;
}

File: python/하노이의탑.py
def solution(n, start=1, to=3, aux=2, answer=[]):
  if n == 1:
    answer.append([start, to])
    return answer
    
  solution(n-1, start, aux, to, answer)
  answer.append([start, to])
  solution(n-1, aux, to, start, answer)
  return answer

print(solution(3))
